### ISSUE TITLE
Auto mark fast follows

### ISSUE THREAD

**Issue by mbostock** [observablehq/plot#1252]
Fast follows from #1238:

- [ ] Pass through other channel options, _e.g._ thresholds for binning
- [ ] Pass through other mark options, _e.g._ strokeWidth
- [ ] How to specify _z_ for multi-series line or area marks?
- [ ] How to limit and sort bar charts?

**Comment by tophtucker** [comment]
Maybe lower-priority, but not to be forgotten: 
- [ ] "Shorthand" (make it work for passing arrays of primitives)
- [ ] Better default unconfigured state for `Plot.auto(aapl).plot()` than an error ([notes](https://observablehq.com/d/bde38bbf32f27fec))

**Comment by tophtucker** [comment]
It’s a little hard to say why, but this should be lines, not rects:

<img width="893" alt="image" src="https://user-images.githubusercontent.com/841829/216683850-ce052aad-3710-41dc-adf4-9a705e83f7b4.png">

Is it because date_of_birth is continuous, albeit not monotonic, and reduce is something other than count? Implicitly stacking means feels wrong.

**Comment by tophtucker** [comment]
An explicit size reducer should stop us from setting an implicit y reducer:

<img width="575" alt="image" src="https://user-images.githubusercontent.com/841829/216684494-f5688a5a-ab6e-412b-b560-e6e1c9b3666d.png">

**Comment by mbostock** [comment]
> reduce is something other than count

Yes, we should only implicitly stack when the reduce is count, not when it’s something else. You can always add counts but adding means/medians/etc. often doesn’t make sense.

**Comment by Fil** [comment]
_sum_ is also additive :-) I'm adding it to the TODO.

**Comment by tophtucker** [comment]
Setting size to an ordinal dimension shows nothing. That makes sense, since `Plot.dot(data, {x: "culmen_depth_mm", r: "sex"}).plot()` shows nothing either, and I'm not sure what it'd mean… maybe size by the value's index in the domain? But it feels broken when I'm trying every combination and every other one works. Could also throw an error.

<img width="691" alt="image" src="https://user-images.githubusercontent.com/841829/217330422-085aabe3-45ee-4f4f-a930-a04312b8bfe9.png">

**Comment by mbostock** [comment]
> Setting size to an ordinal dimension shows nothing.

Related #493; the data here is being filtered because the _size_ channel implicitly coerces to numbers producing NaN.

**Comment by tophtucker** [comment]
Should we throw an error when a column isn't found in the data?

Two notable cases. You might've made a typo (as described in Robert's Levenshtein distance comment https://github.com/observablehq/plot/issues/493#issuecomment-1196115315, though imo that's overkill):

```
Plot.auto(aapl, {x: "close"}).plot()
```

Or you might've changed the data source, invalidating your column selections:

```
Plot.auto(olympians, {x: "Close"}).plot()
```

Other marks don't do this, so I imagine you've considered it and decided against it, but it might fit with Plot.auto's more aggressive and opinionated mandate.

**Comment by tophtucker** [comment]
"Error: implicit ordinal position domain has more than 10,000 values" is the most common (only?) error I've been seeing, which makes me itch for limit/sort. Maybe the channel options object could have a limit property; undefined would default to ~20; if null, it wouldn't limit; if it limits, it always sorts? `{x: {value: "nationality", limit: 100}}`

**Comment by mbostock** [comment]
Closing this umbrella issue. Please re-open individual feature issues as needed for better tracking.
